### PR TITLE
docs: Generate documentation with sphinx

### PR DESCRIPTION
**Pull request by Melkor333**
I introduced **sphinx docs** and just used our Adfinis template (FWIW this can probably be replaced by plain sphinx+configuration at some point, see my [PR and its discussion](https://github.com/adfinis/adsy-sphinx-template/pull/2)).
    - Please [download the latest artifact](https://github.com/adfinis/pyaptly/actions/runs/12410382358/artifacts/2342434566) and take a look at it. I'm quite happy with how it looks right now (not perfect, but IMO good enough to be merged)
- Added a **docs task in the Makefile**
    - Yes I manually install md-click because it's kinda abandoned. I'd love to revive it in the Adfinis Namespace, but if you devs don't have a use for it, let's just stick to the quick-fix for now.
- **usage docs** with md-click. This generates multiple files which I merge to make it more convenient
    - I had to rename `@cli.command` to `@pyaptly.command` in `cli.py` to make the generated docs more comprehensible.
    - I merge the generated files and do some adjustment of the headers in the Makefile
- **code docs** with `Autodocs` (not very useful but only took like 5 mins... :) )
- A `jsonschema` of the **configuration file** at `pyaptly/config.schema.json`
   - I use it to generate the docs
   - I also added a validation function to the `main.py` which uses the jsonschema.
- My linter did some things in `publish.py` and `test_publish.py` which I added in a separate commit

### CODE REVIEW

**Comment by Melkor333** [comment]
This should also close #113 when finished.

**Comment by Melkor333** [comment]
Got internal feedback that maybe gitbook or vuebook might be a "more modern" approach than going back to sphinx... Since everything is markdown the only missing piece would be the pydocs which could probably be converted to md with something like [pydoc-markdown](https://pypi.org/project/pydoc-markdown/)

**Review by winged — CHANGES_REQUESTED** [review]
Awesome work! I only have a little nitpick, otherwise LGTM!

File: pyaptly/main.py [hunk 1]
```diff
@@ -3,15 +3,26 @@
...
 
 from . import command, custom_logger, mirror, publish, repo, snapshot, util
 
 _logging_setup = False
 
 
 lg = logging.getLogger(__name__)
+__location__ = os.path.realpath( os.path.join(os.getcwd(), os.path.dirname(__file__)))
+with open(__location__ + '/config.schema.json', 'r') as file:
+    jsonschema=json.load(file)
```
[the remark this answers is not in the document]
Comment (Melkor333) [reply pyaptly/main.py]: Thanks! I knew this code's gonna require some changes 😄

**Review by winged — APPROVED** [review]

**Pull request merged**